selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/AddOrImportProjectFormTest.java

```diff
@@ -0,0 +1,504 @@
+    newWorkspace.waitRamValue("dev-machine", 3.0);
+
+    newWorkspace.clickOnAddOrImportProjectButton();
+    newWorkspace.waitAddOrImportFormOpened();
+    newWorkspace.clickOnSampleCheckbox(SPRING_SAMPLE_NAME);
+    newWorkspace.waitSampleCheckboxEnabled(SPRING_SAMPLE_NAME);
+    newWorkspace.clickOnAddButtonInImportProjectForm();
+    newWorkspace.waitProjectTabAppearance(SPRING_SAMPLE_NAME);
+    newWorkspace.clickOnBottomCreateButton();
+    newWorkspace.waitWorkspaceCreatedDialogIsVisible();
+    newWorkspace.closeWorkspaceCreatedDialog();
+    newWorkspace.waitWorkspaceCreatedDialogDisappearance();
+    workspaceOverview.checkNameWorkspace(TEST_BLANK_WORKSPACE_NAME);
+
+    seleniumWebDriver.navigate().back();
+
+    prepareJavaWorkspaceAndOpenCreateDialog(TEST_JAVA_WORKSPACE_NAME);
+
+    newWorkspace.clickOnEditWorkspaceButton();
+    workspaceOverview.checkNameWorkspace(TEST_JAVA_WORKSPACE_NAME);
+
+    seleniumWebDriver.navigate().back();
+
+    prepareJavaWorkspaceAndOpenCreateDialog(TEST_JAVA_WORKSPACE_NAME_EDIT);
+
+    newWorkspace.waitWorkspaceCreatedDialogIsVisible();
+    newWorkspace.clickOnOpenInIDEButton();
+
+    testWorkspaceServiceClient.waitStatus(
+        TEST_JAVA_WORKSPACE_NAME_EDIT, defaultTestUser.getName(), RUNNING);
+
+    seleniumWebDriverHelper.switchToIdeFrameAndWaitAvailability();
+
+    projectExplorer.waitProjectExplorer();
+    projectExplorer.waitItem(SPRING_SAMPLE_NAME);
+    projectExplorer.expandPathInProjectExplorerAndOpenFile(
+        SPRING_SAMPLE_NAME + "/src/main/java/org.eclipse.che.examples", "GreetingController.java");
+    editor.waitActive();
+    editor.waitTextIntoEditor(EXPECTED_TEXT_IN_EDITOR);
+  }
+
+  private void waitAllCheckboxesDisabled() {
+    newWorkspace
+        .getSamplesNames()
+        .forEach(sampleName -> newWorkspace.waitSampleCheckboxDisabled(sampleName));
+  }
+
+  private void waitAllCheckboxesEnabled() {
+    newWorkspace
+        .getSamplesNames()
+        .forEach(sampleName -> newWorkspace.waitSampleCheckboxEnabled(sampleName));
+  }
+
+  private void clickOnEachCheckbox() {
+    newWorkspace
+        .getSamplesNames()
+        .forEach(sampleName -> newWorkspace.clickOnSampleCheckbox(sampleName));
+  }
+
+  private void checkProjectTabAppearanceAndFields(
+      String tabName, String expectedDescription, String expectedUrl) {
+    newWorkspace.waitProjectOptionsForm();
+    newWorkspace.waitProjectNameFieldValueInProjectOptionsForm(tabName);
+    newWorkspace.waitDescriptionFieldValueInProjectOptionsForm(expectedDescription);
+    newWorkspace.waitRepositoryUrlFieldValueInProjectOptionsForm(expectedUrl);
+    newWorkspace.waitRemoveButtonInProjectOptionsForm();
+    newWorkspace.waitCancelButtonDisablingInProjectOptionsForm();
+    newWorkspace.waitSaveButtonDisablingInProjectOptionsForm();
+  }
+
+  private void prepareJavaWorkspaceAndOpenCreateDialog(String workspaceName) {
+    newWorkspace.waitPageLoad();
```

Comment:
It's hard to understand test scenario here. Please, divide separate steps by empty rows.